applications/GeoMechanicsApplication/tests/cpp_tests/test_backward_euler_T_scheme.cpp

```diff
@@ -49,8 +49,71 @@ KRATOS_TEST_CASE_IN_SUITE(BackwardEulerScheme_UpdatesVariablesDerivatives_WhenPr
     scheme.Predict(model_part, dof_set, A, Dx, b);
 
     constexpr double expected_dt_temperature = 0.25;
-    KRATOS_EXPECT_DOUBLE_EQ(p_node->FastGetSolutionStepValue(DT_TEMPERATURE),
+    KRATOS_EXPECT_DOUBLE_EQ(p_node->FastGetSolutionStepValue(DT_TEMPERATURE, 0),
                             expected_dt_temperature);
 }
 
+KRATOS_TEST_CASE_IN_SUITE(InitializeBackwardEulerTScheme_SetsTimeFactors, KratosGeoMechanicsFastSuite)
+{
+    BackwardEulerTScheme<SparseSpaceType, LocalSpaceType> scheme;
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+
+    model_part.AddNodalSolutionStepVariable(TEMPERATURE);
+    model_part.AddNodalSolutionStepVariable(DT_TEMPERATURE);
+
+    constexpr double delta_time = 3.0;
+    model_part.GetProcessInfo()[DELTA_TIME] = delta_time;
+
+    scheme.Initialize(model_part);
+
+    KRATOS_EXPECT_TRUE(scheme.SchemeIsInitialized())
+    KRATOS_EXPECT_DOUBLE_EQ(
+        model_part.GetProcessInfo()[DT_TEMPERATURE_COEFFICIENT], 1.0 / delta_time);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ForMissingNodalDof_CheckBackwardEulerTScheme_Throws, KratosGeoMechanicsFastSuite)
+{
+    BackwardEulerTScheme<SparseSpaceType, LocalSpaceType> scheme;
+
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+    model_part.AddNodalSolutionStepVariable(TEMPERATURE);
+    model_part.AddNodalSolutionStepVariable(DT_TEMPERATURE);
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(scheme.Check(model_part),
+                                      "missing TEMPERATURE dof on node ")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ForMissingDtTemperatureSolutionStepVariable_CheckBackwardEulerTScheme_Throws,
+                          KratosGeoMechanicsFastSuite)
+{
+    BackwardEulerTScheme<SparseSpaceType, LocalSpaceType> scheme;
+
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+    model_part.AddNodalSolutionStepVariable(TEMPERATURE);
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
+    p_node->AddDof(TEMPERATURE);
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        scheme.Check(model_part),
+        "DT_TEMPERATURE variable is not allocated for node 0")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ForMissingTemperatureSolutionStepVariable_CheckBackwardEulerTScheme_Throws,
+                          KratosGeoMechanicsFastSuite)
+{
+    BackwardEulerTScheme<SparseSpaceType, LocalSpaceType> scheme;
+
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
```

Comment:
You may want to add variable `DT_TEMPERATURE` (to be consistent with test case `ForMissingWaterPressureSolutionStepVariable_CheckBackwardEulerPwScheme_Throws`):
```suggestion
    model_part.AddNodalSolutionStepVariable(DT_TEMPERATURE);
    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
```

Reply:
I removed this line due to the review comment of WPK, which I agree with. I prefer to remove the same line from the Pw version (and the newmark T/Pw equivalents)